pkgs/by-name/ky/kyua/kyua-check-hook.sh

```diff
@@ -0,0 +1,12 @@
+kyuaCheckHook() {
+    # Kyua expects to save test results in ~/.kyua/store
+    HOME=$TMPDIR kyua test
+}
+
+if [ -z "${dontKyuaCheck-}" ]; then
+    if [ -z "${checkPhase-}" ]; then
```

Comment:
If checkPhase is set, the hook shouldn’t run. I can change this to follow the pattern Ninja uses if that would make things clearer. I can’t remember off hand what I patterned this off.

Reply:
I changed it to a distinct phase and have it set the phase to `kyuaCheckPhase` if `checkPhase` is not already defined (following Ninja’s pattern except renaming things to `Kyua`). This is untested because I don’t want to lose today’s build progress ….

Reply:
But what is the reasoning for not running kyuaCheckHook as long as the checkPhase is set to anything?

Reply:
That seems to be the standard practice for phase hooks. If the user declares an explicit phase (e.g., `checkPhase`), presumably they want full control over anything.